float8_experimental/float8_linear_utils.py

```diff
@@ -322,3 +328,34 @@ def inner_func():
     for child in fp8_layers:
         # Set a flag to signal amaxes/scales are ready
         child.amax_and_scale_synced = True
+
+
+def precompute_float8_amax(module: nn.Module) -> None:
```

Comment:
@weifengpy do you plan / want to use compile on this, and are there any gaps around here that you think would be good to prioritize on the compile side?

This is mostly just me remembering @awgu mention a while ago that he thought compile added noticeable runtime overhead, and I can't remember if it was for this specific case. If it is, and we think compiling this code would be useful, I can prioritize looking into the runtime overhead.

Reply:
Hi @bdhirsh, I plan to polish and land this PR without compile next week to conclude H1. most importantly add `_pre_computed_amax` to flatten/unflatten

Reducing runtime overhead from `torch.compile` is still meaningful since we want torch.compile(fp8 casting) in FSDP2 pre-forward hooks. would it be helpful if I work on a mini repro with profiler traces? Want to unblock you in the short-term

Reply:
If you have a mini repro showing bad runtime overheads with compile, that would be great!

Reply:
Hi @bdhirsh , I have created a repro https://github.com/pytorch/pytorch/issues/129457 . I highlighted extra cpu overhead and gpu time for torch.compile(mode="reduce-overhead")